parse/lex_test.go

```diff
@@ -550,9 +578,23 @@ func assertItems(t *testing.T, want []item, l *lexer) {
 
 	logItems := make([]func(), 0, len(want))
 
-	for _, wantItem := range want {
+	i := 0
```

Comment:
Collecting items first would simplify the flow and make assertion more reliable.